colcon_core/shell/sh.py

```diff
@@ -26,6 +26,8 @@ class ShShell(ShellExtensionPoint):
     FORMAT_STR_USE_ENV_VAR = '${name}'
     FORMAT_STR_INVOKE_SCRIPT = 'COLCON_CURRENT_PREFIX="{prefix}" ' \
         '_colcon_prefix_sh_source_script "{script_path}"'
+    FORMAT_STR_CLEANUP_TRAILING_SEPARATORS = 'echo ${name} | ' \
+        'grep -q "{value}$" && export {name}=${{{name}%?}}'
```

Comment:
Simplify to not check the value, only care about if there's a trailing separator.